Geometry/TrackerGeometryBuilder/src/trackerHierarchy.cc

```diff
@@ -1,39 +1,32 @@
-#include "DataFormats/SiPixelDetId/interface/PXBDetId.h"
-#include "DataFormats/SiPixelDetId/interface/PXFDetId.h"
-#include "DataFormats/SiStripDetId/interface/TIBDetId.h"
-#include "DataFormats/SiStripDetId/interface/TIDDetId.h"
-#include "DataFormats/SiStripDetId/interface/TOBDetId.h"
-#include "DataFormats/SiStripDetId/interface/TECDetId.h"
+#include "Geometry/TrackerGeometryBuilder/interface/trackerHierarchy.h"
 
 #include<string>
 #include<vector>
 
-std::string trackerHierarchy(unsigned int rawid) {
+std::string trackerHierarchy(const TrackerTopology* tTopo, unsigned int rawid) {
   DetId id(rawid);
   int subdetid = id.subdetId();
   switch (subdetid) {
     
     // PXB
   case 1:
```

Comment:
While we're editing this, could we change these into PixelSubdetector::PixelBarrel, PixelSubdetector::PixelEndcap, StripSubdetector::TIB, StripSubdetector::TID, StripSubdetector::TOB, and StripSubdetector::TOB?